scripts/test_aics_cluster.py

```diff
@@ -0,0 +1,207 @@
+        signal.signal(signal.SIGALRM, signal_handler)
+        signal.alarm(timeout)
+        print(f"Will wait for: {timeout} seconds")
+
+        # Actual wait for workers
+        client.cluster.scale(n_workers)
+        client.wait_for_workers(n_workers)
+
+        # Close signal
+        signal.alarm(0)
+        print("Successfully waited for workers!")
+    except TimeoutError:
+        log.error("Timed out wait for workers check")
+
+
+def run_iteration(file: Path, save_path: Path) -> Path:
+    # Read image
+    img = AICSImage(file)
+
+    # Select middle slice of structure channel
+    data = img.get_image_data(
+        "YX", S=0, T=0, C=img.get_channel_names().index("structure"), Z=img.size_z // 2,
+    )
+
+    # Write out image as png
+    imwrite(save_path, data)
+
+    return save_path
+
+
+def run_image_read_checks(client: Client, n_workers: int):
+    # Spawn workers
+    client.cluster.scale(n_workers)
+
+    # Get test image path
+    source_image = Path(__file__).parent / "resources" / "example.ome.tiff"
+
+    # Run check iterations
+    print("Starting read image iterations...")
+    with DistributedHandler(client.cluster.scheduler_address) as handler:
+        handler.batched_map(
+            run_iteration,
+            [source_image for i in range(10000)],
+            [source_image.parent / f"{i}.png" for i in range(10000)],
+        )
+
+
+def deep_cluster_check(
+    cores_per_worker: int,
+    memory_per_worker: str,
+    n_workers: int,
+    timeout: int = 600,  # seconds
+):
+    print("Checking wait for workers...")
+    print("Spawning SLURMCluster...")
+    client = spawn_cluster(
+        cluster_type="wait_for_workers",
+        cores_per_worker=cores_per_worker,
+        memory_per_worker=memory_per_worker,
+        n_workers=n_workers,
+    )
+    run_wait_for_workers_check(client=client, timeout=timeout, n_workers=n_workers)
+
+    print("Wait for workers check done. Tearing down cluster.")
+    client.shutdown()
+    client.close()
+    print("=" * 80)
+
+    print("Waiting a bit for full cluster teardown")
+    time.sleep(120)
+
+    print("Checking IO iterations...")
+    print("Spawning SLURMCluster...")
+    client = spawn_cluster(
+        cluster_type="io_iterations",
+        cores_per_worker=cores_per_worker,
+        memory_per_worker=memory_per_worker,
+        n_workers=n_workers,
+    )
+    # Log time duration
+    start = time.perf_counter()
+    run_image_read_checks(client=client, n_workers=n_workers)
+    print(f"IO checks completed in: {time.perf_counter() - start} seconds")
+
+    print("IO iteration checks done. Tearing down cluster.")
+    client.shutdown()
+    client.close()
+    print("=" * 80)
+
+    print("All checks complete")
+
+
+########################################################################################
+# Actual tests
+
+
+@pytest.mark.parametrize("cores_per_worker", [1, 2, 4])
+@pytest.mark.parametrize("n_workers", [12, 24, 32, 64, 128])
```

Comment:
128 too small?  do you want to use this to find the breaking point?

Reply:
I don't particularly want to find the breaking point but I did change these in a recent commit to be from 12 - 256 with less midpoints.